fdbclient/include/fdbclient/BlobCipher.h

```diff
@@ -50,6 +52,55 @@
 #define AES_256_KEY_LENGTH 32
 #define AES_256_IV_LENGTH 16
 
+class BlobCipherMetrics {
+public:
+	static BlobCipherMetrics* getInstance() {
+		static BlobCipherMetrics* instance = nullptr;
+		if (instance == nullptr) {
+			instance = new BlobCipherMetrics;
+		}
+		return instance;
+	}
+
+	enum UsageType : int {
+		UNKNOWN = 0,
+		TLOG,
+		KV_MEMORY,
+		KV_REDWOOD,
+		BLOB_GRANULE,
+		BACKUP,
+		MAX,
+	};
+
+	struct CounterSet {
+		Counter encryptCPUTimeNS;
+		Counter decryptCPUTimeNS;
+		LatencySample getCipherKeysLatency;
+		LatencySample getLatestCipherKeysLatency;
+
+		CounterSet(CounterCollection& cc, std::string name);
+	};
+
+	static CounterSet& counters(UsageType t) { return getInstance()->counterSets[int(t)]; }
+
+private:
+	BlobCipherMetrics();
+
+	CounterCollection cc;
```

Comment:
For better understanding: any gain keeping `CounterCollection` as `private`? Especially given all individual metrics counters are `public`.

Reply:
Because callers don't ever need to access `cc`. But making it public doesn't hurt.